metaci/build/flows.py

```diff
@@ -1,17 +1,16 @@
-from cumulusci.core.flows import BaseFlow
+from cumulusci.core.flowrunner import FlowCallback
 from django.utils import timezone
 
 from .models import FlowTask
 
 
-class MetaCIFlow(BaseFlow):
-    """ An implementation of BaseFlow that logs task execution to the database. """
+class MetaCIFlowCallback(FlowCallback):
+    """ An implementation of FlowCallback that logs task execution to the database. """
 
-    def __init__(self, *args, **kwargs):
-        self.buildflow_id = None
-        super(MetaCIFlow, self).__init__(*args, **kwargs)
+    def __init__(self, buildflow_id):
+        self.buildflow_id = buildflow_id
     
-    def _pre_task(self, task):
+    def pre_task(self, task):
```

Comment:
the first arg to these is now the StepSpec, not the BaseTask, so, step.path, step.step_num etc...

Reply:
Hmm. I can mostly work with that but it means we don't have the final initialized options.

Reply:
i mean, we can change it, but thats what the code does now so this will fail wth attributeerrors.

hmm.